### ISSUE TITLE
Compilation errors when using multiple pragmas and solc >= 0.7.4

### ISSUE THREAD

**Issue by thegostep** [sc-forks/solidity-coverage#611]
<img width="526" alt="image" src="https://user-images.githubusercontent.com/15959632/106397697-a957f180-63dc-11eb-84f8-afa6db4b7d48.png">

**Comment by cgewecke** [comment]
@thegostep 

Are you using the latest solidity-coverage version? (0.7.14)

Can you show the relevant solidity files?

**Comment by thegostep** [comment]
as far as I can tell, it is related to using `pragma abicoder v2;` in solo 0.7

**Comment by thegostep** [comment]
Example file:

```solidity
// SPDX-License-Identifier: GPL-3.0-only
pragma solidity 0.7.6;
pragma abicoder v2;

import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
import {IERC721} from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
import {IGeyser} from "../Geyser.sol";
import {IFactory} from "../Factory/IFactory.sol";
import {IUniversalVault} from "../UniversalVault.sol";

contract MockStakeHelper {
    function flashStake(
        address geyser,
        address vault,
        address recipient,
        uint256 amount,
        bytes calldata lockPermission,
        bytes calldata unstakePermission
    ) external {
        IGeyser(geyser).stake(vault, amount, lockPermission);
        IGeyser(geyser).unstakeAndClaim(vault, recipient, amount, unstakePermission);
    }

    function stakeBatch(
        address[] calldata geysers,
        address[] calldata vaults,
        uint256[] calldata amounts,
        bytes[] calldata permissions
    ) external {
        for (uint256 index = 0; index < vaults.length; index++) {
            IGeyser(geysers[index]).stake(vaults[index], amounts[index], permissions[index]);
        }
    }
}
```

**Comment by thegostep** [comment]
> Are you using the latest solidity-coverage version? (0.7.14)

yessir

**Comment by cgewecke** [comment]
Ah it might be a bug introduced in that version - some logic was added to support file level function definitions and it might not handle multiple pragmas correctly.

Could you see if downgrading to 0.7.13 fixes?

**Comment by thegostep** [comment]
> Could you see if downgrading to 0.7.13 fixes?

Indeed - downgrading fixed it!

**Comment by cgewecke** [comment]
Ok sweet - thanks for reporting.

**Comment by giosdas** [comment]
Hello
To help people to found this issue and save time I report here the full stacktrace in text format:

	CompileError: /home/user/Projects/smart-contracts/.coverage_contracts/TokenContract.sol:3:1: DeclarationError: Function with same name and parameter types defined twice.
	function c_0x56fb40c0(bytes32 c__0x56fb40c0) pure {}
	^--------------------------------------------------^
	/home/user/Projects/smart-contracts/.coverage_contracts/TokenContract.sol:6:1: Other declaration is here:
	function c_0x56fb40c0(bytes32 c__0x56fb40c0) pure {}
	^--------------------------------------------------^

	Compilation failed. See above.
		at run (/home/user/Projects/smart-contracts/node_modules/truffle/build/webpack:/packages/compile-solidity/run.js:53:1)
		at Object.sourcesWithDependencies (/home/user/Projects/smart-contracts/node_modules/truffle/build/webpack:/packages/compile-solidity/index.js:121:58)
		at all (/home/user/Projects/smart-contracts/node_modules/truffle/build/webpack:/packages/compile-solidity/index.js:61:1)
		at /home/user/Projects/smart-contracts/node_modules/truffle/build/webpack:/packages/workflow-compile/legacy/index.js:98:31
		at async Promise.all (index 0)
		at Object.compile (/home/user/Projects/smart-contracts/node_modules/truffle/build/webpack:/packages/workflow-compile/legacy/index.js:50:1)
		at plugin (/home/user/Projects/smart-contracts/node_modules/solidity-coverage/plugins/truffle.plugin.js:97:5)

Regards

**Comment by cgewecke** [comment]
@giosdas Thanks for pinging. Just patched this in `0.7.15`. Please lmk if you're still seeing issues after upgrading
